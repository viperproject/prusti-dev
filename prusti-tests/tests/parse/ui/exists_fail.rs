// compile-flags: -Pprint_desugared_specs=true -Pprint_typeckd_specs=true -Pno_verify=true -Phide_uuids=true
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"
// normalize-stdout-test: "\[[a-z0-9]{4}\]::" -> "[$(CRATE_ID)]::"

use prusti_contracts::*;

#[requires(exists)]
fn test1() {}

#[requires(exists())]
fn test2() {}

#[requires(exists(|))]
fn test3() {}

#[requires(exists(||) 1+1)]
fn test4() {}

#[requires(exists(|a, b| true))]
fn test5() {}

#[requires(exists(||) || exists(||))]
fn test6() {}

#[requires(exists(|| 1+1 == 1+1, triggers=[1]))]
fn test7() {}

#[requires(exists(|| true, triggers=[(1,2), 1]))]
fn test8() {}

#[requires(exists(|| true, triggers=1))]
fn test9() {}

#[requires(exists(||))]
fn test10() {}

#[requires(exists(|| 1+1 == 1+1))]
fn test11() {}

#[requires(exists(||, triggers=[]))]
fn test12() {}

#[requires(exists(|| 1+1 == 1+1, triggers=[(1,)]))]
fn test13() {}

fn main() {}
