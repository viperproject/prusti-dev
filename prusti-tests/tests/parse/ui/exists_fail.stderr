error: expected `(`
 --> $DIR/exists_fail.rs:8:12
  |
8 | #[requires(exists)]
  |            ^^^^^^

error: expected `|`
  --> $DIR/exists_fail.rs:11:18
   |
11 | #[requires(exists())]
   |                  ^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:14:19
   |
14 | #[requires(exists(|))]
   |                   ^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:17:19
   |
17 | #[requires(exists(||) 1+1)]
   |                   ^^

error: expected `:`
  --> $DIR/exists_fail.rs:20:21
   |
20 | #[requires(exists(|a, b| true))]
   |                     ^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:23:19
   |
23 | #[requires(exists(||) || exists(||))]
   |                   ^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:26:19
   |
26 | #[requires(exists(|| 1+1 == 1+1, triggers=[1]))]
   |                   ^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:29:19
   |
29 | #[requires(exists(|| true, triggers=[(1,2), 1]))]
   |                   ^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:32:19
   |
32 | #[requires(exists(|| true, triggers=1))]
   |                   ^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:35:19
   |
35 | #[requires(exists(||))]
   |                   ^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:38:19
   |
38 | #[requires(exists(|| 1+1 == 1+1))]
   |                   ^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:41:19
   |
41 | #[requires(exists(||, triggers=[]))]
   |                   ^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:44:19
   |
44 | #[requires(exists(|| 1+1 == 1+1, triggers=[(1,)]))]
   |                   ^^

error: aborting due to 13 previous errors

